src/metpy/testing.py

```diff
@@ -215,4 +215,12 @@ def wrapper(*args, **kwargs):
     return dec
 
 
+def xfail_dask(array, reason):
+    """Xfail a test if `array` is a `dask.array.Array`."""
+    dask_array = pytest.importorskip('dask.array', reason='dask.array not available')
+
+    if not isinstance(array, xr.DataArray) and isinstance(array.m, dask_array.Array):
+        pytest.xfail(reason=reason)
```

Comment:
The downside to this approach (calling `pytest.xfail()`) instead of using the `@pytest.mark.xfail` is that my read on the [documentation](https://docs.pytest.org/en/latest/how-to/skipping.html#xfail-mark-test-functions-as-expected-to-fail) is that calling `pytest.xfail()` means we'll never get an XPASS.

I'm digging in to see if there is a better way we can structure some of this with pytest.